### ISSUE TITLE
Undo-redo: Mutating XMLElement attributes before deleting the element causes the attribute to be permanently lost

### ISSUE THREAD

**Issue by aartrost** [yjs/yjs#447]
**Checklist**

* [x] Are you reporting a bug? Use github issues for bug reports and feature requests. For general questions, please use https://discuss.yjs.dev/
* [x] Try to report your issue in the correct repository. Yjs consists of many modules. When in doubt, report it to https://github.com/yjs/yjs/issues/

**Describe the bug**
Undo/redo loses XMLElement attributes when mutating an attribute right before deleting it in the same transaction.

**To Reproduce**

```javascript
import * as Y from "yjs";

const origin = "origin";
const doc = new Y.Doc();
const fragment = doc.getXmlFragment("t" + Date.now());
const undoManager = new Y.UndoManager(fragment, {
  captureTimeout: 0,
  trackedOrigins: new Set([origin]),
});

// create element
doc.transact(() => {
  const e = new Y.XmlElement("test-node");
  e.setAttribute("a", "1");
  e.setAttribute("b", "2");
  fragment.insert(fragment.length, [e]);
}, origin);

console.log('STEP 1: Create test node with "a" and "b" attributes. Result:', fragment.toString())

// change attribute "b" and delete test-node
doc.transact(() => {

  const e = fragment.get(0);
  e.setAttribute("b", "3");
  fragment.delete(e);

}, origin);

console.log('STEP 2: Mutate test node "b" attribute, then delete the node. Result:', fragment.toString())

// undo
undoManager.undo();

console.log('STEP 3: Undo the deletion. Only the unchanged attribute "a" is still present:', fragment.toString())
```

Output: 
```
STEP 1: Create test node with "a" and "b" attributes. Result: <test-node a="1" b="2"></test-node>
STEP 2: Mutate test node "b" attribute, then delete the node. Result: 
STEP 3: Undo the deletion. Only the unchanged attribute "a" is still present: <test-node a="1"></test-node>
```

**Expected behavior**
Expecting the output after undo to be `<test-node a="1" b="2"></test-node>` as this is the output from the first transaction

**Environment Information**
 - Browser / Chrome
 - Yjs [v13.5.40](https://github.com/yjs/yjs/releases/tag/v13.5.40)

**Comment by dmonad** [comment]
Thank you! I fixed the issue in `v13.5.41`.

This should also close https://github.com/yjs/yjs/issues/443

**Comment by aartrost** [comment]
Perfect, this worked like a charm. It also resolved another issue we were seeing where XMLElements would lose all of their children after being reordered (a delete + insert op which would explain why it was related).

Thank you!
